src/models/Deployment.ts

```diff
@@ -88,6 +91,7 @@ export class Deployment implements IDeployment {
     this.workPoolName = deployment.workPoolName
     this.enforceParameterSchema = deployment.enforceParameterSchema
     this.pullSteps = deployment.pullSteps
+    this.can = deployment.can ?? createDeploymentCan()
```

Comment:
Wonder if there's some way we can safeguard agains this being used accidentally in cloud. Perhaps we could add an `if cloud` type check and throw an error if the dummy checks are used?

Reply:
is there some cloud boolean somewhere already? 

I feel _slightly_ not concerned about this only because in cloud, you shouldn't have to create a deployment can anywhere - (🤞 atleast right now...) it should just be on deployment instantiation. 

i feel like the root of the smell is really that we're not least privilege by default - ~~we're instead open access by default so where it _feels_ like ideally we should always fallback to no priv,~~
hmm i guess to your point, maybe the constructor _should_ require the can? whereas I let it be optional for easier compatibility...

Reply:
I realized when reviewing the cloud pr for this that this wouldn't be possible anyway. We don't instantiate these classes anywhere outside of the service methods and in cloud the service method automatically sets this. So I'm not worried about this anymore.

Reply:
I don't feel strongly about requiring the can be passed in but it would provide a safe guard if someone does `new Deployment()` somewhere. Might be best?